trieTools/zeroBalanceSystemAccountChecker/extraTokensCrossChecker.go

```diff
@@ -0,0 +1,159 @@
+package main
+
+import (
+	"errors"
+
+	"github.com/ElrondNetwork/elrond-go-core/core"
+	"github.com/tidwall/gjson"
+)
+
+const (
+	maxRequestsRetrial = 10
+	multipleSearchBulk = 10000
+)
+
+type tokenBalancesGetter interface {
+	getBalance(address, token string) (string, error)
+}
+
+type elasticMultiSearchClient interface {
+	GetMultiple(index string, requests []string) ([]byte, error)
+}
+
+type extraTokensChecker struct {
+	nftBalancesGetter tokenBalancesGetter
+	elasticClient     elasticMultiSearchClient
+}
+
+func newExtraTokensCrossChecker(client elasticMultiSearchClient, nftBalancesGetter tokenBalancesGetter) (crossTokenChecker, error) {
```

Comment:
unit tests? Why not?

Reply:
Most likely in a future PR, since this one is already pretty big